halpybot/packages/board/board.py

```diff
@@ -133,37 +187,27 @@ def __contains__(self, key: typing.Union[str, int]) -> bool:
             return key in self._cases_by_id
         return False
 
-    async def add_case(self, client, platform: Platform, system) -> Case:
+    async def add_case(self, client: str, platform: Platform, system: str) -> Case:
         """Create a new Case given the client name"""
         async with self._modlock:
             new_id = self.open_rescue_id
             case = Case(
                 board_id=new_id, client_name=client, platform=platform, system=system
             )
-            self._last_case_time = now(tz="utc")
+            self._update_last_index()
             if case.client_name in self._case_alias_name:
                 raise ValueError("Case with Client Name Already Exists")
 
             self._cases_by_id[new_id] = case
             self._case_alias_name[client] = new_id
             return case
 
-    @asynccontextmanager
-    async def mod_case(self, case: Case):
+    async def mod_case(self, case_id: int, new_case: Case):
         """
-        Modify an existing case'
-        TODO: Is this thing on?
+        Modify an existing case
         """
-        async with self._modlock:
-            current_case = case.board_id
-            current_client = case.client_name
-            del self._cases_by_id[current_case]
-            del self._case_alias_name[current_client]
-            try:
-                yield case
-            finally:
-                self._cases_by_id[current_case] = case
-                self._case_alias_name[current_client] = current_case
+        new_case = evolve(new_case, updated_time=now(tz="UTC"))
+        self._cases_by_id[case_id] = new_case
```

Comment:
good progress on this.


I do wonder if we could simplify usage something to the effect of
```py
@functools.wraps(evolve)
async def mod_case(self, * , case_id, **kwargs):
    new_case = evolve(self._cases_by_id[case_id], updated_time=now(tz="UTC"), **kwargs)
```

Reply:
Only one way to find out :)